cloudwatch_logs_common/src/utils/cloudwatch_facade.cpp

```diff
@@ -147,6 +149,37 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CreateLogGroup(
   return status;
 }
 
+Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CheckLogGroupExist(
+  const std::string & log_group)
+{
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors status = CW_LOGS_SUCCEEDED;
+
+  Aws::CloudWatchLogs::Model::DescribeLogGroupsRequest describe_log_group_request;
+  describe_log_group_request.SetLogGroupNamePrefix(log_group.c_str());
+
+  const auto & response = this->cw_client_.DescribeLogGroups(describe_log_group_request);
+  if (!response.IsSuccess()) {
+    status = CW_LOGS_LOG_GROUP_NOT_FOUND;
+    AWS_LOGSTREAM_INFO(__func__, "Log group named: " << log_group
+            << " does not exist. Error message: "
+            << response.GetError().GetMessage() << ", with error code: "
+            << static_cast<int>(response.GetError().GetErrorType()));
+  }
+
+  if (CW_LOGS_SUCCEEDED == status) {
```

Comment:
Why not just using an `else` for the `if (!response.IsSuccess())`, instead of this additional `if? I think the control flow would be more clear, the current version depends on the assignment `Aws::CloudWatchLogs::ROSCloudWatchLogsErrors status = CW_LOGS_SUCCEEDED` at the start of the method.